butane_cli/src/lib.rs

```diff
@@ -0,0 +1,265 @@
+use std::{fs::File, io::Write, path::PathBuf};
+
+use butane::migrations::{
+    copy_migration, FsMigrations, MemMigrations, Migration, MigrationMut, Migrations, MigrationsMut,
+};
+use butane::query::BoolExpr;
+use butane::{db, db::Connection, db::ConnectionMethods, migrations};
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+
+pub type Result<T> = std::result::Result<T, anyhow::Error>;
+
+#[derive(Serialize, Deserialize, Default)]
+pub struct CliState {
+    embedded: bool,
+}
+impl CliState {
+    pub fn load() -> Result<Self> {
+        let path = base_dir()?.join("clistate.json");
+        let file = File::open(path);
+        match file {
+            Ok(file) => Ok(serde_json::from_reader(file)?),
+            Err(_) => Ok(CliState::default()),
+        }
+    }
+
+    pub fn save(&self) -> Result<()> {
+        let path = base_dir()?.join("clistate.json");
+        let file = File::create(path)?;
+        serde_json::to_writer(file, &self)?;
+        Ok(())
+    }
+}
+
+pub fn default_name() -> String {
+    Utc::now().format("%Y%m%d_%H%M%S%3f").to_string()
+}
+
+pub fn init(name: &str, connstr: &str) -> Result<()> {
+    if db::get_backend(name).is_none() {
+        eprintln!("Unknown backend {name}");
```

Comment:
Should we switch to some sort of logging facade (log or trace) here? I'm not sure direct printing makes sense for library usage